evalml/model_understanding/graphs.py

```diff
@@ -437,11 +437,12 @@ def partial_dependence(pipeline, X, feature, grid_resolution=100):
     elif isinstance(pipeline, evalml.pipelines.RegressionPipeline):
         pipeline._estimator_type = "regressor"
     pipeline.feature_importances_ = pipeline.feature_importance
-    avg_pred, values = sk_partial_dependence(pipeline, X=X, features=[feature], grid_resolution=grid_resolution)
-
-    # Delete scikit-learn attributes that were temporarily set
-    del pipeline._estimator_type
-    del pipeline.feature_importances_
+    try:
```

Comment:
If we have to repeat this pattern in the future, it'd be cool to use a context manager!